### ISSUE TITLE
Request: separate version of the plugin that is just the grammar

### ISSUE THREAD

**Issue by bolinfest** [gandm/language-babel#64]
On Nuclide, we have a whole process for all things transpilation, so we really just want proper syntax highlighting for ES6+ files. Trying to navigate the settings that come with this package is intimidating/confusing.

Specifically, looking at the Settings, I don't know what "Transpile on Save" means. Transpile to where? Is this package writing files? If so, where are these files being written?

Basically, my out of the box experience with this package was: "Wow this syntax highlighting is great...Wait, how do I get rid of that popup?"

I'm sure that the transpilation stuff is valuable for a lot of workflows, but I was under the impression that most langauge-XXX packages for Atom were syntax highlighters for XXX.

**Comment by bolinfest** [comment]
Or how about just changing `transpileOnSave` to default to `false`?

https://github.com/gandm/language-babel/blob/master/lib/config.coffee#L4

**Comment by bolinfest** [comment]
After looking through the code, I see that I can just uncheck the "Transpile On Save" checkbox to get what I want. However, I don't feel comfortable encouraging others en masse (say anyone who works on Nuclide) to use this if I know they're all going to hit this speedbump with the popup.

**Comment by gandm** [comment]
I thought the readme was quite explanatory but I'm sorry if it is confusing. If you want nuclide to do everything then as you rightly say just disable 'transpile on save' which in fact just runs the transpiler as a check for syntax but does nothing else. This was written before Nuclide and I just needed a quick way to get ES2015 running.  The problem I have is that I have no idea if anyone is using this package nor what bits they use. Some people have asked me to disable the grammar for JSX, others want to extend the language-babel setting to a project file directory level. 

I could change the default to disable 'transpile on save' and I don't think that would cause any existing users a problem. I may well do this on the next publish.

**Comment by bolinfest** [comment]
@gandm Thanks for the quick reply: defaulting to disable 'transpile on save' would certainly do it for me. Then no one encounters unexpected pop-ups after installing this package for the first time. Those who need functionality beyond syntax highlighting can easily dig into the settings to find it.

**Comment by gandm** [comment]
Relevant part of readme is 
Use Cases

Pure JavaScript with no EcmaScript 2015 requirements.
Just use the Grammar to color code the scripts and turn off the package setting  Transpile On Save .

**Comment by bolinfest** [comment]
Perfect -- thank you so much!

**Comment by bolinfest** [comment]
Can we reopen this? It seems like more code is being added to this package to add more support for the transpiler. This is adversely affecting the load time for this package, which is a problem for those of us who just want syntax highlighting.

If things were refactored so that anything the transpilation needs were lazy-loaded, that would also help, but the eager loading is a killer.

**Comment by gandm** [comment]
As I mentioned I will split the grammar so I'll reopen this. I don't think the time taken to load the app is much to do with eager vs. lazy loading of dependencies; although that would help. I just made my app have an en empty activate and deactivate and removed all requires from all coffeescripts and it was still slow. Removing the grammar helps a lot - as shown in the second shot! 

Provided the cache is up to date the grammar seems to be about 70% of the overall load time. Third screen shot group.

Empty activate/deactivate & no requires
![image](https://cloud.githubusercontent.com/assets/2313237/10017045/47c2a576-6123-11e5-853a-ef6c535cc50d.png)

![image](https://cloud.githubusercontent.com/assets/2313237/10016913/316ef294-6122-11e5-8266-c1a3d8d0362c.png)

No Grammar
![image](https://cloud.githubusercontent.com/assets/2313237/10016969/badd633a-6122-11e5-97e7-d5673f1f7e6a.png)

App back to normal and with grammar
![image](https://cloud.githubusercontent.com/assets/2313237/10017753/01006be0-6129-11e5-98c6-96c8d37dd300.png)

**Comment by bolinfest** [comment]
https://github.com/gandm/language-babel/pull/85 explains it all.

If you're still serious about publishing a version of this package that is only the grammar, then that's great and I'll drop the issue. If not, I'll create my own internal fork that does this.

**Comment by zertosh** [comment]
Hey @gandm and @bolinfest, I just had a quick chat with @sebmck, and we think there's a place under the babel org for the grammar only. How about we open up babel/babel-atom (a sibling to babel/babel-sublime) to host & promote the grammar (without the transpile stuff) as the official babel one? What do you think @gandm?

**Comment by gandm** [comment]
@zertosh , @bolinfest , @sebmck I don't have a problem, anyone can take the grammar and do with it whatever they want. If it meets your needs; at least until the AST can be used as grammar; then fine. 
You may want to check out subtleGradients grammar, language-javascript-jsx, reading one of his issues he indicates he has a new grammar but hasn't released it yet. It may be better than mine. 

Let me know if you proceed so that I don't create my own separate grammar.
